### PR TITLE
Apartment improvements oracle migration

### PR DESCRIPTION
**Pull request by ranta**
# Hitas Pull Request

# Description

- In oracle migration use the newest calculation with any improvements at all
- Move certain improvements values to the apartment 'Additional Work During Construction'
- Refactoring

## Pull request checklist

Check the boxes for each DoD items that has been completed:

- **Frontend**
    - [ ] Changes have been tested
- **Backend**
    - [x] Changes have been tested
    - [ ] Automatic tests has been added
    - [ ] OpenAPI definitions have been updated
    - [ ] Database migrations will work in test environment
    - [x] Oracle migration has been updated
    - [ ] initial.json has been updated to work with migrations

## Test plan

Trust me bro

## Tickets

This pull request resolves all or part of the following ticket(s): HT-372, HT-373

### CODE REVIEW

File: backend/hitas/management/commands/hitasmigrate.py [hunk 1]
```diff
@@ -25,8 +27,24 @@ def add_arguments(self, parser: CommandParser) -> None:
...
+        parser.add_argument(
+            "--oracle-instantclient-path",
+            default="Downloads/instantclient_19_8",
+            help="Oracle instantclient location, MacOS only (default: 'Downloads/instantclient_19_8').",
+        )
 
     def handle(self, *args, **options) -> None:
+        if sys.platform.startswith("darwin"):  # MacOS
+            try:
+                lib_dir = os.path.join(os.environ.get("HOME"), options["oracle_instantclient_path"])
```
Comment (matti-lamppu) [inline backend/hitas/management/commands/hitasmigrate.py:42]: [Pathlib](https://docs.python.org/3/library/pathlib.html) is the way to go nowadays, but this is fine as well, no need to change it.

File: backend/hitas/oracle_migration/runner.py [hunk 2]
```diff
@@ -494,101 +501,149 @@ def create_apartments(
...
+            if cpi_awdc["date"] > mpi_awdc["date"]:
+                awdc = sum_cpi
+            else:
+                awdc = sum_mpi
+        else:
+            # Unable to convert improvements automatically.
+            # Print out and add them as normal improvements to be handled manually later.
+            bulk_cpi.extend(cpi_awdc["improvements"])
+            bulk_mpi.extend(mpi_awdc["improvements"])
+            print(
```
Comment (matti-lamppu) [inline backend/hitas/oracle_migration/runner.py:631]: Might want to add some title to the print like "Unhandled construction time improvement, adding as a regular improvement:". Also, would a multiline string look cleaner?

**Review by matti-lamppu — APPROVED** [review]
Looks good, I trust you with the logic 💪

File: backend/hitas/oracle_migration/runner.py (hunk 2 above)
Comment (ranta) [reply backend/hitas/oracle_migration/runner.py:631]: Yep, changed this to multiline string :)

**Pull request merged**